cpp/src/barretenberg/honk/proof_system/work_queue.hpp

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "barretenberg/honk/transcript/transcript.hpp"
+#include "barretenberg/plonk/proof_system/proving_key/proving_key.hpp"
+#include <cstddef>
+#include <memory>
+
+namespace proof_system::honk {
+
+// Currently only one type of work queue operation but there will likely be others related to Sumcheck
+enum WorkType { SCALAR_MULTIPLICATION };
+
+template <typename Params> class work_queue {
```

Comment:
Is this the same sort of `Params` expected by some other class? Would be helpful to document that here if so.

Reply:
That's right. Comment/TODO added for a bit more clarity